### ISSUE TITLE
TempData with cookies doesn't work on ManageController.LinkLoginCallback

### ISSUE THREAD

**Issue by rmarskell** [aspnet/templating#164]
In `ManageController` in the `LinkLoginCallback` action, [setting](https://github.com/aspnet/templating/blob/09161305bb26edc371ca4504d2fecc9b13da09a7/src/Microsoft.DotNet.Web.ProjectTemplates/content/StarterWeb-CSharp/Areas/Identity/Controllers/ManageController.cs#L285) `StatusMessage = "The external login was added.";`  doesn't persist to the `RedirectToAction( nameof( ExternalLogins ) )` if using cookies for Temp Data. Changing the Temp Data storage to session state makes it work. I'm guessing it has something to do with setting cookies on the redirect, but I'm not sure.

Repro:

1. Create a new ASP.NET Core Web Application with Individual User Accounts
2. Add an external provider (I tested with Twitter)
3. Register as a local user (don't use external log in)
4. Go to /Manage/ExternalLogins and add Twitter
5. When you're eventually redirected back to /Manage/ExternalLogins, no status message is shown because the cookie-based temp data isn't persisted
6. Enable session storage for TempData and repeat above steps to see correct message.

Packages:
Microsoft.AspNetCore.All - Version="2.0.0"
Microsoft.EntityFrameworkCore.Tools - Version="2.0.1"
Microsoft.VisualStudio.Web.CodeGeneration.Design - Version="2.0.1"

Tested on Chrome 62.

Also, I don't know if this is technically an issue in the templates or another repo, so sorry if this is the wrong place.

**Comment by Eilon** [comment]
@jbagga can you take a look to see what's going on here?

**Comment by jbagga** [comment]
I was able to repro with a 2.0 MVC app but not in dev

**Comment by jbagga** [comment]
For Chrome 63:

It may be because in 2.0 the server sends the CookieTempDataProvider cookie as a Response Cookie and the redirect to ExternalLogins does not send the same as the Request cookie.

In dev, the CookieTempDataProvider received is sent back as a Request cookie and TempData appears as expected.

This does not repro in Firefox though. 

I am investigating further if Google Chrome has an issue with setting cookies on Redirects

cc @kichalla

**Comment by jbagga** [comment]
It has been fixed in 2.1 and there is a workaround for running into this issue. See [this comment](https://github.com/aspnet/Mvc/issues/7190#issuecomment-355639083)

Add this to your Startup.cs
```
services.Configure<CookieTempDataProviderOptions>( options =>
{
	options.Cookie.SameSite = SameSiteMode.Lax; // By default this is set to 'Strict'.
} );
```
